### ISSUE TITLE
Speed up inference for constant prompt

### ISSUE THREAD

**Issue by fefespn** [IDEA-Research/GroundingDINO#152]
If I want to run G-DINO multiple times on one prompt alone,
can I save sometime in the inference somhow ?
or How could I distill/decrease the model wights/inference time when I know I have 1 prompt ?

To retrain a Yolo over the output of G-DINO is good idea, but my database is small so
the yolo didn't learn very will !

**Comment by rentainhe** [comment]
> If I want to run G-DINO multiple times on one prompt alone, can I save sometime in the inference somhow ? or How could I distill/decrease the model wights/inference time when I know I have 1 prompt ?
> 
> To retrain a Yolo over the output of G-DINO is good idea, but my database is small so the yolo didn't learn very will !

I think auto-distill can help you to distill a yolo like model using Grounding-DINO, you can check it here: https://github.com/autodistill/autodistill

**Comment by fefespn** [comment]
Thanks @rentainhe , that is what I am doing (but without the library). 
Still this need images dataset to train on the distilled model.
I am searching something like to take just the knowledge of G-Dino on that prompt and to through away the unwanted weights.
so by that I will not lower the accuracy and with no need for a dataset to train on.

**Comment by lyy1988323** [comment]
with the constant prompt, you really could save the prompt encode time. you'll see the [demo](https://github.com/IDEA-Research/GroundingDINO/blob/6c27bc76b9c7252c201e0b8c680d7349a7a0aaf8/demo/inference_on_a_image.py#L112) 
`   

    # filter output
    if token_spans is None:
        logits_filt = logits.cpu().clone()
        boxes_filt = boxes.cpu().clone()
        filt_mask = logits_filt.max(dim=1)[0] > box_threshold
        logits_filt = logits_filt[filt_mask]  # num_filt, 256
        boxes_filt = boxes_filt[filt_mask]  # num_filt, 4

        # get phrase
        tokenlizer = model.tokenizer
        tokenized = tokenlizer(caption)
        # build pred
        pred_phrases = []
        for logit, box in zip(logits_filt, boxes_filt):
            pred_phrase = get_phrases_from_posmap(logit > text_threshold, tokenized, tokenlizer)
            if with_logits:
                pred_phrases.append(pred_phrase + f"({str(logit.max().item())[:4]})")
            else:
                pred_phrases.append(pred_phrase)`
